### ISSUE TITLE
Using cvxpy.min / cvxpy.max

### ISSUE THREAD

**Issue by Hadi2525** [cvxpy/cvxpy#1672]
I am trying to work around the following gp in using cvxpy.
```
import cvxpy as cp

alpha = 0.3
lamda_1 = 0.5
lamda_2 = 1.2

mu1 = cp.Variable(pos=True)
mu2 = cp.Variable(pos=True)
p_ev_max = 50
eps = 0.05
S1 = cp.Variable(pos=True)
S2 = cp.Variable(pos=True)
S = cp.Parameter(pos=True)
S.value = 200
# time resolution is 15 mins
obj = cp.Minimize(1/(mu1 - lamda_1)+1/(mu2 - lamda_2))
constraints = [
    mu1 == cp.min(p_ev_max, (1+eps)*S1),
    mu2 == cp.min(p_ev_max, (1+eps)*S2),
    S == S1 + S2,
]
prob = cp.Problem(objective= obj, constraints = constraints)
prob.solve(gp=True, requires_grad=True)
```
But I get the following error:
```
Traceback (most recent call last):
  File "<input>", line --, in <module>
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/expressions/expression.py", line 661, in __lt__
    raise NotImplementedError("Strict inequalities are not allowed.")
NotImplementedError: Strict inequalities are not allowed.
```
I am not sure why this error is raised as the description is not related to the constraint. 

I looked at the documentation to find an example to be able to handle this but would appreciate if this feature can be included.

**Comment by rileyjmurray** [comment]
Hi @Hadi2525, that error is strange indeed. What version of cvxpy are you running?

**Comment by Hadi2525** [comment]
Hi @rileyjmurray , it is 1.1.18.

```
Traceback (most recent call last):
  File "<input>", line 3, in <module>
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/atoms/min.py", line 29, in __init__
    super(min, self).__init__(x, axis=axis, keepdims=keepdims)
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/atoms/axis_atom.py", line 36, in __init__
    super(AxisAtom, self).__init__(expr)
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/atoms/atom.py", line 50, in __init__
    self.validate_arguments()
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/atoms/axis_atom.py", line 60, in validate_arguments
    if self.axis is not None and self.axis > self.args[0].ndim:
  File "/Users/hadirohani/Library/Python/3.8/lib/python/site-packages/cvxpy/expressions/expression.py", line 680, in __gt__
    raise NotImplementedError("Strict inequalities are not allowed.")
NotImplementedError: Strict inequalities are not allowed.

```
This is the complete error.

**Comment by Hadi2525** [comment]
Thank you for mentioning that. In the case that this is not a DGP problem, is it because the min constraints are not of DGP type? Or should I remove the `gp` argument from the solver?

**Comment by rileyjmurray** [comment]
The constraints don't meet the criteria for being DGP. At first look, it seems they are very nonconvex (by which I mean, I think the problem will remain nonconvex under typical changes of variables). I recommend you familiarize yourself with the DGP documentation: https://www.cvxpy.org/tutorial/dgp/index.html or to ask for help on our Discord.

For the purposes of this issue, this is resolved by #1670.
